woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -677,39 +677,37 @@ def test_underlying_index(sample_df):
         pytest.xfail('Setting underlying index is not supported with Koalas input')
 
     specified_index = pd.Index
+    unspecified_index = pd.RangeIndex
 
     schema_df = sample_df.copy()
     schema_df.ww.init(index='full_name')
     assert schema_df.index.name is None
     assert (schema_df.index == ['Mr. John Doe', 'Doe, Mrs. Jane', 'James Brown', 'Ms. Paige Turner']).all()
     assert type(schema_df.index) == specified_index
 
-    # --> add back when schema updates are implemented
-    # schema = Schema(sample_df.copy())
-    # schema = schema.set_index('full_name')
-    # assert (schema._dataframe.index == schema.to_dataframe()['full_name']).all()
-    # assert schema._dataframe.index.name is None
-    # assert type(schema._dataframe.index) == specified_index
-    # assert type(schema.to_dataframe().index) == specified_index
+    schema_df.ww.set_index('full_name')
+    assert (schema_df.index == schema_df['full_name']).all()
+    assert schema_df.index.name is None
+    assert type(schema_df.index) == specified_index
```

Comment:
I'm not sure where it belongs exactly, but I think we should make sure somewhere that when we set the underlying dataframe index that the index column also remains present in the DataFrame columns as this is the behavior we expect. Maybe I missed it, but I'm not seeing that check.

Reply:
I think this test is a fine place to add that check. Makes sense to add